<!DOCTYPE html>
<html>
<head>
    <title>L1000CDS²</title>
	<meta charset='UTF-8'>
	i
    <link rel="shortcut icon" href="images/sigine.ico" type="image/x-icon" />
	<meta content='width=device-width, initial-scale=1' name='viewport'/>
	<link rel='stylesheet' type='text/css' href='css/main.css'/>
	<link rel='stylesheet' type='text/css' href='css/open-sans.css'/>
</head>
<body>
    <div id='sidebar'>
        <ul>
            <li><a href='#summary'>Summary</a></li>
            <li><a href='#L1000Data'>L1000 Data</a></li>
            <li><a href='#tutorial'>Tutorial</a></li>
            <ul id='sideBarSub'>
                <li><a href='#search'>Search</a></li>
                <li><a href='#example'>Example</a></li>
                <li><a href='#EBOV'>EBOV</a></li>
                <li><a href='#aggravate'>Mimic/Reverse</a></li>
                <li><a href='#diseases'>Diseases</a></li>
                <li><a href='#table'>Table</a></li>
                <li><a href='#share'>Share</a></li>
            </ul>
            <li><a href='#api'>API</a></li>
            <li><a href='#contact'>Contact</a></li>
        </ul>
    </div>
    <div id='content'>
        <h2 id='title'>
            <img id='logo' src='images/sigine.png'/>
            <span>L1000CDS²</span>
        </h2>
        <h4 id='subtitle'>An <span class="highlight lightning-fast">ultra-fast</span> LINCS <span class="highlight">L1000 </span><span class="highlight">C</span>haracteristic <span class="highlight">D</span>irection <span class="highlight">S</span>ignature<span class="highlight"> S</span>earch Engine</h4>
        <section id='summary'>
            <h3 class='first'>Summary</h3>
            <p>L1000CDS² is a LINCS L1000 characteristic direction signature search engine to find consensus signatures that match users’ input gene lists or input signatures. The underlying dataset is the <a target="_blank" href="http://www.lincscloud.org/">LINCS L1000</a> small molecule expression profiles generated at the Broad Institute by the <a target="_blank" href="http://lincscloud.org/about/cmap-team/">Connectivity Map team</a>. The differentially expressed (DE) genes of these profiles were calculated using the <a target="_blank" href="http://maayanlab.net/CD/">characteristic direction</a> method. Depending on the user’s input, L1000CDS² uses either a gene-set method or cosine distance method to run the search. When up/down gene lists are submitted to L1000CDS², the search engine compares the input gene lists to the DE genes computed from the LINCS L1000 data and descriptive information of the top 50 matched signatures is returned. When a signature is submitted to L1000CDS² in the format of "gene symbol, expression value", the search engine calculates a cosine distance between the input signature and every characteristic direction signature in the database, and the top 50 signatures of either the largest (reverse mode) or the smallest (mimic mode) cosine distances are returned. L1000CDS² leverages the efficiency of matrix operations to perform the search and finishes a query against more than 20,000 signatures in less than 0.1sec using the gene-set method or less than 3sec using the cosine distance method. The app is developed by the <a href='http://icahn.mssm.edu/research/labs/maayan-laboratory' target='_blank'>Ma'ayan Laboratory</a> at the <a href='http://icahn.mssm.edu/' target='_blank'>Icahn School of Medicine at Mount Sinai</a> for the <a href='http://www.lincs-dcic.org/' target='_blank'>BD2K-LINCS-DCIC</a> and the <a href='http://commonfund.nih.gov/idg/fundedresearch' target='_blank'>KMC-IDG </a>NIH funded projects.</p>
        </section>
        <section id='L1000Data'>
            <h3>L1000 mRNA Expression Data</h3>
            <p>The L1000 mRNA gene-expression dataset is generated as part of the <a href="http://www.lincsproject.org/">Library of Integrated Network-based Cellular Signatures (LINCS) project</a>, a <a href="https://commonfund.nih.gov/">Common Fund</a> program run by NIH. LINCS aims to systematically profile the molecular and phenotypical outcomes of agent perturbed human cells. The variety of agents includes chemical compounds, different micro-environments, endogenous ligands, gene knockdown and overexpression. The L1000 dataset comprises of over 1 million gene expression profiles of chemically or genetically perturbed human cell-lines. The L1000 technology developed at Broad Institute measures only ~1000 genes in an experiment and estimates the expression of the rest of the transcriptome using an empirical model. In this way, the technology significantly reduces the cost and makes large-scale gene expression profiling possible. L1000CDS² currently includes a subset of the chemically perturbed gene expression profiles, specifically the profiles in the CPC and CPD peturbagen groups. After averaging replicates and removing insignificant signatures, we remain with 22,926 signatures in the L1000CDS² database.</p>
        </section>
       
        <section id='tutorial'>
            <h3>Tutorial</h3>
            <div class="col-left">
                <h5 id="search">Search</h5>
                <p>The entry point into L1000CDS² is to paste up/down gene lists into the up/down gene text boxes or paste a signature into the up gene text box. The search button will only become enabled when both up/down genes text boxes are filled by gene lists or when the up gene textbox is filled with a signature. Click the examples to see the formats of gene-set and signature inputs. Click the Search button and the information for the top 50 signatures will be displayed in a <a href="#table">table</a> on the right side.</p>
                <h5 id="example">Example</h5>
                <p>Clicking the Gene-set Example button will fill in an example of up/down gene lists in the text boxes for a demo search using the gene-set method. Clicking the Signature Example button will fill in an example of a signature in the up gene text box for a demo of searching with the cosine distance method.</p>
                <h5 id="EBOV">EBOV</h5>
                <p>Clicking the EBOV Signatures button will open a table of three EBOV signatures at three time points. Select a signature by a single click and click the Search button to perform a cosine distance search. The selected signature will be automatically filled in the up gene text box.</p>
                <h5 id="aggravate">Mimic/Reverse</h5>
                <p>In the mimic mode for the gene-set search, the input up genes are intersected with the up genes of the gene expression profiles in the L1000CDS² database, and the input down genes are intersected with the down genes for each entry in the database. In the reverse mode for the gene-set search, the input up genes are intersected with the down genes, and the input down genes are intersected with the up genes. When a cosine distance search is performed, the top 50 signatures will be those of the smallest cosine distances from the input signature in the mimic mode, or those with the largest cosine distances in the reverse mode. <strong>The default mode is reverse.</strong></p>

            </div>
            <div class="col-right">
                <img id="fig1" src='images/search.png'/>
                <p class='caption'>Fig. 1 Screenshot of the input text boxes, the search button, the example button, the EBOV signatures button, the diesease signatures button  and the mimic/reverse slider.</p>
            </div>

            <h5 id="diseases">Diseases</h5>
            <p>Clicking on the Diseases Signature button will open a table of disease names and their tissue types including the GEO ID refering to the original study for the disease signature (Fig 2). The table is searchable by disease name or tissue type and can be sorted by either column. Each row represents a differential expression signature of a disease and consists of the diferentially expressed genes and their expression values in the disease compared with the relevant normal tissue. These signatures were calculated from gene expression data deposited in GEO.  Clicking on a row will select that disease. Clicking on the search button will fill in the signature of the selected disease in the up gene text boxes and perform the search for small molecules at the same time. </p>
            <img id="fig2" class="diseases-figure" src="images/diseases.png"/>
            <p class="caption">Fig 2 Screenshot of the disease table</p>

            <h5 id="table">Table</h5>
            <p>The search results are rendered as a paginated table with 12 entries on each page (Fig 3). Each entry provides five pieces of information about the signature: score, perturbation, cell-line, dose and time point. The score is the overlap between the input DE genes and the signature DE genes divided by the effective input. The effective input is the length of the intersection between the input genes and the L1000 genes since some input lists contain genes that are not present in the L1000 dataset. For the cosine distance search, the score is the cosine distance between the input signature and the L1000 characteristic direction signatures. The consensus signatures are sorted by their scores in descending order. The perturbation column shows the names of the chemical perturbations. Clicking on a name will direct the user to a web page within <a target="_blank" href="http://life.ccs.miami.edu/life/">LIFE</a> that catalogs detailed information about the chemical compound. There is also a signature column in the table. Clicking the download button in that column will download all the information about a signature as a JSON file. </p>
            <img id="fig3" src='images/table.png'/>
            <p class='caption'>Fig. 3 Screenshot of the paginated results table and the share button.</p>
            <h5 id="share">Share</h5>
            <p>After performing a search, a share button (Fig 3) will appear above the results table at the the top right corner. Clicking on the button will show a unique URL for sharing the search results.</p>
            
        </section>
        
        <section id='api'>
            <h3>API</h3>
            <p><strong>[POST] http://amp.pharm.mssm.edu/L1000CDS2/query</strong></p>
            <!-- http://localhost:8083/g2e/full?accession=GDS5077&platform=GPL10558&control=GSM1071454-GSM1071455&experimental=GSM1071457-GSM1071456 -->
            <table>
                <caption>Payload for the Gene-set Search <span> (content-type: application/json)</span></caption>
                <tr>
                    <td>upGenes</td>
                    <td>[String]</td>
                    <td>An array of up-regulated Entrez gene symbols.</td>
                </tr>
                <tr>
                    <td>dnGenes</td>
                    <td>[String]</td>
                    <td>An array of down-regulated Entrez gene symbols.</td>
                </tr>
                <tr>
                    <td>aggravate</td>
                    <td>Boolean</td>
                    <td>True to perform the search in mimic mode and false in reverse mode.</td>
                </tr>
                 <tr>
                    <td>searchMethod</td>
                    <td>String</td>
                    <td>"geneSet"</td>
                <tr>
            </table>
            <table>
                <caption>Payload for the Cosine distance Search <span> (content-type: application/json)</span></caption>
                <tr>
                    <td>input</td>
                    <td>Object</td>
                    <td>A wrapper for input gene symbols and expression values.</td>
                </tr>
                <tr>
                    <td>input.genes</td>
                    <td>[String]</td>
                    <td>An array of Entrez gene symbols.</td>
                </tr>
                <tr>
                    <td>input.vals</td>
                    <td>[Number]</td>
                    <td>An array of differential expression values, e.g. t-test statistics.</td>
                </tr>
                <tr>
                    <td>aggravate</td>
                    <td>Boolean</td>
                    <td>True to perform the search in mimic mode and false in reverse mode.</td>
                </tr>
                 <tr>
                    <td>searchMethod</td>
                    <td>String</td>
                    <td>"CD"</td>
                <tr>
            </table>
            <table>
                <caption>Response <span>(JSON)</span></caption>
                <tr>
                    <td>shareId</td>
                    <td>String</td>
                    <td>Unique ID for sharing the search results</td>
                </tr>
                <tr>
                    <td>topMeta</td>
                    <td>[Object]</td>
                    <td>Descriptive information of the top 50 consensus signatures</td>
                </tr>
                <tr>
                    <td>topMeta[ ].score</td>
                    <td>Double</td>
                    <td>Score</td>
                </tr>
                <tr>
                    <td>topMeta[ ].cell_id</td>
                    <td>String</td>
                    <td>Cell-line</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_desc</td>
                    <td>String</td>
                    <td>Perturbation name</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_id</td>
                    <td>String</td>
                    <td>Unique identifier for a perturbation</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_time</td>
                    <td>String</td>
                    <td>Time point</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_time_unit</td>
                    <td>String</td>
                    <td>Time point unit</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_dose</td>
                    <td>String</td>
                    <td>Dose</td>
                </tr>
                 <tr>
                    <td>topMeta[ ].pert_dose_unit</td>
                    <td>String</td>
                    <td>Dose unit</td>
                </tr>
                 <tr>
                    <td>topMeta[ ].sig_id</td>
                    <td>String</td>
                    <td>Unique identifier for a signature</td>
                </tr>
            </table>

            <table ><caption>P<span>ython</span> Example</caption></table>
            <pre>
import requests
import json
url = 'http://amp.pharm.mssm.edu/L1000CDS2/query'
# gene-set search example
payload = {"upGenes":["KDM5A","EGR1","RELB"],
"dnGenes":["USP22","PHGDH","HADH"],"aggravate":True,"searchMethod":"geneSet"}
headers = {'content-type':'application/json'}
r = requests.post(url,data=json.dumps(payload),headers=headers)
print(r.json)
# cosine distance search example
payload = {"input"{"genes":["DDIT4","HIG2","FLT1","ADM","SLC2A3","ZNF331"],
"vals":[9.97,10.16,7.66,17.80,20.29,15.22]},"aggravate":True,"searchMethod":"CD"}
headers = {'content-type':'application/json'}
r = requests.post(url,data=json.dumps(payload),headers=headers)
print(r.json)
            </pre>
        </section>
        <section id='contact'>
            <h3>Contact</h3>
            <p>Please contact Avi Ma’ayan and Qiaonan Duan for comments, suggestions, and support: <a href='mailto:avi.maayan@mssm.edu'>avi.maayan@mssm.edu</a>, <a href='mailto:qiaonan.duan@mssm.edu'>qiaonan.duan@mssm.edu</a>.</p>
        </section>
    </div>
</body>
</html>
