<!DOCTYPE html>
<html>
<head>
    <title>L1000CDS²</title>
	<meta charset='UTF-8'>
	i
    <link rel="shortcut icon" href="images/sigine.ico" type="image/x-icon" />
	<meta content='width=device-width, initial-scale=1' name='viewport'/>
	<link rel='stylesheet' type='text/css' href='css/main.css'/>
	<link rel='stylesheet' type='text/css' href='css/open-sans.css'/>
</head>
<body>
    <div id='sidebar'>
        <ul>
            <li><a href='#summary'>Summary</a></li>
            <li><a href='#L1000Data'>L1000 Data</a></li>
            <li><a href='#tutorial'>Tutorial</a></li>
            <ul id='sideBarSub'>
                <li><a href='#search'>Search</a></li>
                <li><a href='#example'>Example</a></li>
                <li><a href='#EBOV'>EBOV</a></li>
                <li><a href='#aggravate'>Mimic/Reverse</a></li>
                <li><a href='#diseases'>Diseases</a></li>
                <li><a href='#table'>Table</a></li>
                <li><a href='#share'>Share</a></li>
            </ul>
            <li><a href='#api'>API</a></li>
            <li><a href='#contact'>Contact</a></li>
        </ul>
    </div>
    <div id='content'>
        <h2 id='title'>
            <img id='logo' src='images/sigine.png'/>
            <span>L1000CDS²</span>
        </h2>
        <h4 id='subtitle'>A <span class="highlight lightning-fast">lightning-fast</span> LINCS <span class="highlight">L1000 </span><span class="highlight">C</span>haracteristic <span class="highlight">D</span>irection <span class="highlight">S</span>ignature<span class="highlight"> S</span>earch Engine</h4>
        <section id='summary'>
            <h3 class='first'>Summary</h3>
            <p>L1000CDS² is a LINCS L1000 characteristic direction signature search engine to find consensus signatures that match users’ input gene lists. The underlying dataset is the <a target="_blank" href="http://www.lincscloud.org/">LINCS L1000</a> small molecule expression profiles generated at the Broad Institute by the <a target="_blank" href="http://lincscloud.org/about/cmap-team/">Connectivity Map Group</a>. The differentially expressed (DE) genes of these profiles were calculated using <a target="_blank" href="http://maayanlab.net/CD/">characteristic direction</a> method and deposited into a database. When up/down gene lists are submitted to L1000CDS², the search engine compares the input gene lists to the DE genes in the database and returns the descriptive information of the top 50 most consensus signatures. L1000CDS² leverages the efficiency of matrix operation to perform the search and can finish a query against more than 20,000 signatures in <0.1s. The app is being actively developed by the <a href='http://icahn.mssm.edu/research/labs/maayan-laboratory' target='_blank'>Ma'ayan Laboratory</a> at the <a href='http://icahn.mssm.edu/' target='_blank'>Icahn School of Medicine at Mount Sinai</a> for the <a href='http://www.lincs-dcic.org/' target='_blank'>BD2K-LINCS-DCIC</a> and the <a href='http://commonfund.nih.gov/idg/fundedresearch' target='_blank'>KMC-IDG</a>.</p>
        </section>
        <section id='L1000Data'>
            <h3>L1000 Data</h3>
            <p>The L1000 dataset is generated as part of the Library of Integrated Network-based Cellular Signatures (LINCS) project, a Common Fund program run by NIH. LINCS aims to systematically profile the molecular and phonotypical outcomes of agent perturbed cell-lines. The variety of agents includes chemical compounds, gene knockdown and overexpression. The L1000 dataset comprises of about a million gene expression profiles of chemically or genetically perturbed human cell-lines. They were measured using L1000 technology developed at Broad Institute. The L1000 technology measures only ~1000 genes in an experiment and estimates the expression of the rest genome using an empirical model. In this way, the technology significantly reduces the cost and makes large-scale gene expression profiling possible. L1000CDS2 currently includes a subset of the chemically perturbed gene expression profiles, specifically the profiles in the CPC and CPD peturbagen groups. After averaging replicates and removing insignificant ones, we have 22,926 signatures in the L1000CDS² database.</p>
        </section>
       
        <section id='tutorial'>
            <h3>Tutorial</h3>
            <div class="col-left">
                <h5 id="search">Search</h5>
                <p>The entry point of L1000CDS² is to paste up/down gene lists into the up/down gene text boxes. The search button will only be enabled when both up/down genes are filled in. Click the search button and the information of the top 50 signatures will be displayed in a <a href="#table">table</a> on the right side.</p>
                <h5 id="example">Example</h5>
                <p>Clicking the Example button will fill in example up/down gene lists in the text boxes for a demo search.</p>
                <h5 id="EBOV">EBOV</h5>
                <p>Clicking the EBOV Signature button will fill in the up/down gene lists of an EBOV signature in the text boxes for a demo search.</p>
                <h5 id="aggravate">Mimic/Reverse</h5>
                <p>In mimic mode, input up genes are intersected with up genes of the gene expression profiles in the L1000CDS² database and input down genes are intersected with down genes of the gene expression profiles in the database. In reverse mode, input up genes are intersected with down genes of the gene expression profiles in the L1000CDS² database and input down genes are intersected with up genes of the gene expression profiles in the database. <strong>The default mode is reverse.</strong></p>

            </div>
            <div class="col-right">
                <img id="fig1" src='images/search.png'/>
                <p class='caption'>Fig. 1 Screenshot that shows the input text boxes, the search button, the example button, the EBOV signature button, the diesease signatures button  and the mimic/reverse slider.</p>
            </div>

            <h5 id="diseases">Diseases</h5>
            <p>Clicking on the Diseases Signature button will open a table of disease names and their tissue types (Fig 2). The table is searchable by disease name or tissue type and can be sorted by either column. Each row represents a differential expression signature of a disease and consists of 300 up-regulated genes and 300 down-regulated genes. These signatures were calculated from gene expression data deposited in GEO.  Clicking on a row will select that disease. Clicking on the search button will fill in the up/down gene lists of the selected disease in the up/down gene text boxes and perform a search at the same time. </p>
            <img id="fig2" class="diseases-figure" src="images/diseases.png"/>
            <p class="caption">Fig 2 Screenshot that shows the disease table</p>

            <h5 id="table">Table</h5>
            <p>The search results are rendered as a paginated table with 12 entries on each page (Fig 3). Each entry provides 5 pieces of information about the signature: score, perturbation, cell-line, dose and time point. The score is the overlap between the input DE genes and the signature DE genes divided by the effective input. The effective input is the length of intersection between input genes and the L1000 genome since some input genes may not be measured in the L1000 platform. The consensus signatures are sorted by score in descending order. Perturbation column shows the names of the chemical perturbations. Clicking on a name will direct user to a web page provided by the <a target="_blank" href="http://life.ccs.miami.edu/life/">LIFE app</a> that catalogs the detailed information of the chemical compound. There is also a signature column in the table. Clicking a download button in that column will download all the information of a signature as a JSON file. </p>
            <img id="fig3" src='images/table.png'/>
            <p class='caption'>Fig. 3 Screenshot that shows the paginated results table and the share button.</p>
            <h5 id="share">Share</h5>
            <p>After performing a search, a share button (Fig 3) will appear on the top right of the result table. Clicking on the button will show a unique URL for sharing the search results.</p>
            
        </section>
        
        <section id='api'>
            <h3>API</h3>
            <p><strong>[POST] http://amp.pharm.mssm.edu/L1000CDS2/query</strong></p>
            <!-- http://localhost:8083/g2e/full?accession=GDS5077&platform=GPL10558&control=GSM1071454-GSM1071455&experimental=GSM1071457-GSM1071456 -->
            <table>
                <caption>Payload <span>(content-type: application/json)</span></caption>
                <tr>
                    <td>upGenes</td>
                    <td>[String]</td>
                    <td>An array of up-regulated Entrez gene symbols.</td>
                </tr>
                <tr>
                    <td>dnGenes</td>
                    <td>[String]</td>
                    <td>An array of down-regulated Entrez gene symbols.</td>
                </tr>
                <tr>
                    <td>aggravate</td>
                    <td>Boolean</td>
                    <td>True to perform the search in mimic mode and False in reverse mode.</td>
                </tr>
            </table>
            <table>
                <caption>Response <span>(JSON)</span></caption>
                <tr>
                    <td>shareId</td>
                    <td>String</td>
                    <td>Unique ID for sharing the search results</td>
                </tr>
                <tr>
                    <td>topMeta</td>
                    <td>[JSON Object]</td>
                    <td>Descriptive information of the top 50 consensus signatures</td>
                </tr>
                <tr>
                    <td>topMeta[ ].score</td>
                    <td>Double</td>
                    <td>Score</td>
                </tr>
                <tr>
                    <td>topMeta[ ].cell_id</td>
                    <td>String</td>
                    <td>Cell-line</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_desc</td>
                    <td>String</td>
                    <td>Perturbation name</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_id</td>
                    <td>String</td>
                    <td>Unique identifier for a perturbation</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_time</td>
                    <td>String</td>
                    <td>Time point</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_time_unit</td>
                    <td>String</td>
                    <td>Time point unit</td>
                </tr>
                <tr>
                    <td>topMeta[ ].pert_dose</td>
                    <td>String</td>
                    <td>Dose</td>
                </tr>
                 <tr>
                    <td>topMeta[ ].pert_dose_unit</td>
                    <td>String</td>
                    <td>Dose unit</td>
                </tr>
                 <tr>
                    <td>topMeta[ ].sig_id</td>
                    <td>String</td>
                    <td>Unique identifier for a signature</td>
                </tr>
            </table>

            <table ><caption>P<span>ython</span> Example</caption></table>
            <pre>
import requests
import json
url = 'http://amp.pharm.mssm.edu/L1000CDS2/query'
payload = {"upGenes":["KDM5A","EGR1","RELB"],
"dnGenes":["USP22","PHGDH","HADH"],"aggravate":True}
headers = {'content-type':'application/json'}
r = requests.post(url,data=json.dumps(payload),headers=headers)
print(r.json)
            </pre>
        </section>
        <section id='contact'>
            <h3>Contact</h3>
            <p>Please contact Avi Ma’ayan and Qiaonan Duan for comments, suggestions, and support: <a href='mailto:avi.maayan@mssm.edu'>avi.maayan@mssm.edu</a>, <a href='mailto:qiaonan.duan@mssm.edu'>qiaonan.duan@mssm.edu</a>.</p>
        </section>
    </div>
</body>
</html>
